src/gui/powermanagement/powermanagement_x11.cpp

```diff
@@ -75,7 +81,13 @@ void PowerManagementInhibitor::requestIdle()
         return;
 
     m_state = RequestIdle;
-    qDebug("D-Bus: PowerManagementInhibitor: Requesting idle");
+
+    if (m_manager == ManagerType::Systemd)
+    {
+        QDBusUnixFileDescriptor dummy;
+        m_fd.swap(dummy);
```

Comment:
Why not just do the following?
```suggestion
        m_fd.setFileDescriptor(-1);
```

Reply:
And main (maybe stupid) question is why do we need to store it?

Reply:
>Why not just do the following?

That won't do.
https://codebrowser.dev/qt6/qtbase/src/dbus/qdbusunixfiledescriptor.cpp.html#233

>And main (maybe stupid) question is why do we need to store such a descriptor?

Then how do you notify systemd when state is idle? Type `uint` won't work because it doesn't match dbus type, this is enforced by Qt classes.

Reply:
> > Why not just do the following?
> 
> That won't do. https://codebrowser.dev/qt6/qtbase/src/dbus/qdbusunixfiledescriptor.cpp.html#233

Ok.

Reply:
Now I see. It is implemented differently than other managers. There is only one D-Bus call used.